pipeline/src/transformers/addressables/event.ts

```diff
@@ -0,0 +1,73 @@
+import { asDate } from '@prismicio/client';
+
+import {
+  asText,
+  asTitle,
+  isFilledLinkToDocumentWithData,
+  isNotUndefined,
+} from '@weco/content-pipeline/src/helpers/type-guards';
+import { EventPrismicDocument } from '@weco/content-pipeline/src/types/prismic';
+import { ElasticsearchAddressableEvent } from '@weco/content-pipeline/src/types/transformed';
+
+export const transformAddressableEvent = (
+  document: EventPrismicDocument
+): ElasticsearchAddressableEvent | undefined => {
+  const { data, id, tags, uid: documentUid } = document;
+  const format = isFilledLinkToDocumentWithData(data.format)
+    ? asText(data.format.data.title)
+    : undefined;
+
+  const startEnd = data.times
+    .map(t => {
+      return {
+        start: asDate(t.startDateTime),
+        end: asDate(t.endDateTime),
+      };
+    })
+    .filter(d => d.start && d.end)
+    .sort((a, b) => Number(a.start) - Number(b.start));
+
+  const firstStart = startEnd?.at(0)?.start;
+  const lastEnd = startEnd?.at(-1)?.end;
+
+  const times = {
```

Comment:
We return the first possible start dateTime and the last possible end dateTime. We could have turned it into a string here, but we'd lose the ability to present it within `<time>` elements on the frontend, and this way we'll (hopefully) be able to reuse some of our pre-existing date range presentation logic without having to write any new date helper functions (which are already battle-tested in the front-end app)